lib/ecto/adapters/postgres/connection.ex

```diff
@@ -253,10 +253,18 @@ if Code.ensure_loaded?(Postgrex) do
     @impl true
     def explain_query(conn, query, params, opts) do
       {explain_opts, opts} =
-        Keyword.split(opts, ~w[analyze verbose costs settings buffers timing summary]a)
+        Keyword.split(opts, ~w[analyze verbose costs settings buffers timing summary format]a)
+
+      json_format? = {:format, :json} in explain_opts
 
       case query(conn, build_explain_query(query, explain_opts), params, opts) do
-        {:ok, %Postgrex.Result{rows: rows}} -> {:ok, Enum.map_join(rows, "\n", & &1)}
+        {:ok, %Postgrex.Result{rows: rows}} when json_format? ->
+          case Jason.encode(List.flatten(rows)) do
+            {:ok, json} -> {:ok, json }
```

Comment:
```suggestion
            {:ok, json} -> {:ok, json}
```

Reply:
@leandrocp fixed. But I cannot re-request the review, not sure why.